packages/typespec-test/test/batch_modular/generated/typespec-ts/src/BatchService.ts

```diff
@@ -0,0 +1,815 @@
+  JobListPreparationAndReleaseTaskStatusOptions,
+  listPreparationAndReleaseTaskStatus,
+  JobGetTaskCountsOptions,
+  getTaskCounts,
+} from "./api/Job.js";
+import {
+  JobScheduleJobScheduleExistsOptions,
+  jobScheduleExists,
+  JobScheduleDeleteJobScheduleOptions,
+  deleteJobSchedule,
+  JobScheduleGetJobScheduleOptions,
+  getJobSchedule,
+  JobSchedulePatchJobScheduleOptions,
+  patchJobSchedule,
+  JobScheduleUpdateJobScheduleOptions,
+  updateJobSchedule,
+  JobScheduleDisableJobScheduleOptions,
+  disableJobSchedule,
+  JobScheduleEnableJobScheduleOptions,
+  enableJobSchedule,
+  JobScheduleTerminateJobScheduleOptions,
+  terminateJobSchedule,
+  JobScheduleAddJobScheduleOptions,
+  addJobSchedule,
+  JobScheduleListJobSchedulesOptions,
+  listJobSchedules,
+} from "./api/JobSchedule.js";
+import {
+  ApplicationListResult,
+  Application,
+  CustomPagePoolUsageMetrics,
+  BatchPoolListResult,
+  AccountListSupportedImagesResult,
+  PoolNodeCountsListResult,
+  DisableJobOption,
+  BatchJobListResult,
+  BatchJobListPreparationAndReleaseTaskStatusResult,
+  TaskCountsResult,
+  CertificateListResult,
+  NodeFileListResult,
+  BatchJobScheduleListResult,
+  BatchTaskListResult,
+  TaskAddCollectionResult,
+  BatchTaskListSubtasksResult,
+  ComputeNode,
+  ComputeNodeGetRemoteLoginSettingsResult,
+  UploadBatchServiceLogsResult,
+  ComputeNodeListResult,
+  NodeVMExtension,
+  NodeVMExtensionList,
+} from "./api/models.js";
+import {
+  PoolListUsageMetricsOptions,
+  listUsageMetrics,
+  PoolGetAllPoolLifetimeStatisticsOptions,
+  getAllPoolLifetimeStatistics,
+  PoolAddPoolOptions,
+  addPool,
+  PoolListPoolsOptions,
+  listPools,
+  PoolDeletePoolOptions,
+  deletePool,
+  PoolExistsOptions,
+  exists,
+  PoolGetPoolOptions,
+  getPool,
+  PoolPatchPoolOptions,
+  patchPool,
+  PoolDisableAutoScaleOptions,
+  disableAutoScale,
+  PoolEnableAutoScaleOptions,
+  enableAutoScale,
+  PoolEvaluateAutoScaleOptions,
+  evaluateAutoScale,
+  PoolResizeOptions,
+  resize,
+  PoolStopResizeOptions,
+  stopResize,
+  PoolUpdatePropertiesOptions,
+  updateProperties,
+  PoolRemoveNodesOptions,
+  removeNodes,
+} from "./api/Pool.js";
+import {
+  TaskAddTaskOptions,
+  addTask,
+  TaskListTasksOptions,
+  listTasks,
+  TaskAddTaskCollectionOptions,
+  addTaskCollection,
+  TaskDeleteTaskCollectionOptions,
+  deleteTaskCollection,
+  TaskGetTaskCollectionOptions,
+  getTaskCollection,
+  TaskUpdateTaskCollectionOptions,
+  updateTaskCollection,
+  TaskListSubtasksOptions,
+  listSubtasks,
+  TaskTerminateTaskCollectionOptions,
+  terminateTaskCollection,
+  TaskReactivateTaskCollectionOptions,
+  reactivateTaskCollection,
+} from "./api/Task.js";
+import { ClientOptions } from "./common/interfaces.js";
+import { BatchServiceContext } from "./rest/clientDefinitions.js";
+import {
+  PoolStatistics,
+  BatchPool,
+  AutoScaleRun,
+  JobStatistics,
+  BatchJob,
+  Certificate,
+  BatchJobSchedule,
+  BatchTask,
+} from "./rest/models.js";
```

Comment:
Not sure why we have dependency for REST models?

I assume all models are imported from api folder.

Reply:
Good catch fixing this. It looks like the auto fix missing imports from ts-morph is getting the wrong module